Newsletters: add 258 (2023-07-05)

harding opened the pull request:
- [ ] Add lede, maybe add any late news, update releases/RCs, and add topic links @harding 
- [ ] Waiting for confirmation 8 @glozow and @Xekyo 
- [ ] Bitcoin Core 24914 @glozow 
- [ ] Bitcoin Core 27896 @bitschmidty

glozow commented on _posts/en/newsletters/2023-07-05-newsletter.md:
```diff
@@ -0,0 +1,89 @@
+
+- [Bitcoin Core #27896][] removes the experimental system call (syscall) sandbox
+  feature (see [Newsletter #170][news170 syscall]). A [related issue][Bitcoin
+  Core #24771] and follow on comments note the drawbacks to the original
+  approach including maintainability (both of the syscall whitelist and OS
```
```suggestion
  Core #24771] and follow up comments note the drawbacks of the
  feature including maintainability (both of the syscall whitelist and OS
```

glozow commented on _posts/en/newsletters/2023-07-05-newsletter.md:
```diff
@@ -0,0 +1,89 @@
+  feature (see [Newsletter #170][news170 syscall]). A [related issue][Bitcoin
+  Core #24771] and follow on comments note the drawbacks to the original
+  approach including maintainability (both of the syscall whitelist and OS
+  support), better supported alternatives, and considerations whether syscall
```
```suggestion
  support), better-supported alternatives, and considerations about whether syscall
```

Xekyo commented on _includes/specials/policy/en/08-interface.md:
```diff
@@ -0,0 +1,119 @@
+trimmed from commitment transactions. The maximum number of in-flight
+HTLCs cannot exceed 483 per channel, as a commitment transaction with
+more than 483 HTLC outputs would exceed the maximum standard
+size of 400 kWu.
```
```suggestion
size of 100 000 vbytes.
```

Per https://github.com/bitcoinops/bitcoinops.github.io/blob/master/STYLE.md, we should prefer vbytes over weight units.

Xekyo commented on _posts/en/newsletters/2023-07-05-newsletter.md:
```diff
@@ -0,0 +1,89 @@
+Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo], and
+[Bitcoin Inquisition][bitcoin inquisition repo].*
+
+glozow, bitschmidty
```
```suggestion

```

Xekyo reviewed (commented):
Looks good, just got a few nits

harding commented on _includes/specials/policy/en/08-interface.md:
```diff
@@ -0,0 +1,119 @@
+limits, those payments are considered “not enforceable on-chain” and
+trimmed from commitment transactions. The maximum number of in-flight
+HTLCs cannot exceed 483 per channel, as a commitment transaction with
+more than 483 HTLC outputs would exceed the maximum standard
```
> a commitment transaction with more than 483 HTLC outputs would exceed the maximum standard size

I don't believe this is correct.  First, the LN rule is 483 outputs **per party**, so it's 966 outputs plus 2 more outputs for the each party's base commitment outputs.


Second P2WSH 1-input, 968-outputs is  just ~42,000 vbytes.  The reason for the limitation per [BOLT5](https://github.com/lightning/bolts/blob/master/05-onchain.md#rationale-6) is (emphasis mine), "A single transaction that **resolves** all the outputs will be under the standard size limit because of the 483 HTLC-per-party limit".

LN doesn't require all the HTLCs to be resolved in a single transaction---it's just more efficient to do it that way than in individual transactions, and I suppose it also simplifies implementation.  Of note, many (most?) LN implementations currently soft-limit themselves by default to a lower number of per-party pending HTLCs than 483 to reduce their exposure to certain attacks, which I believe is why nobody is pushing to increase the 483 HTLC limit.

Given this section has several other examples of the influence of policy on contracting protocol rules, I think the simplest thing would be to drop this sentence.

glozow replied:
Thanks for catching this! I had a pretty surface-level understanding of this and learned something new today :D

harding commented:
@glozow @Xekyo I put several edits in a0e514d993293692ad8becc495a42acca452e94d ; please feel free to revert / change anything you want different.

Made edits for all actionable feedback (thanks @glozow @Xekyo @bitschmidty !), reviewed contributions (thanks @glozow @Xekyo @bitschmidty, again!) added lede, updated releases/RCs,  and added topic links.  Thanks everyone!

Special thanks to @glozow and @Xekyo for clearly being the primary authors of this week's newsletter, and doing another amazing job on the policy section!

bitschmidty commented:
Squashed and merged, thanks all! 🚀
